cpmpy/solvers/pysat.py

```diff
@@ -258,10 +269,14 @@ def transform(self, cpm_expr):
         cpm_cons = toplevel_list(cpm_expr)
         cpm_cons = decompose_in_tree(cpm_cons)
```

Comment:
`alldifferent` has a specialized decomposition in linearize, so you can also add it to the `supported` set here.